jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -120,9 +120,21 @@ def _on_viewer_added(self, msg):
         self._create_viewer_callbacks(self.app.get_viewer_by_id(msg.viewer_id))
 
     def _on_global_display_unit_changed(self, msg):
-        # eventually should observe change in flux OR angle
-        if msg.axis == "flux":
-            self.image_unit = u.Unit(msg.unit)
+
+        # only concerned with handling unit changes for mouseover in cubeviz,
+        # until implemented in other configs
+        if self.config == 'cubeviz':
+            # even if data loaded is in 'flux' it can be represented as a
+            # per-pixel sb unit, so all cube data will be 'sb'
+            if msg.axis == "sb":
+                image_unit = u.Unit(msg.unit)
+
+                # temporarily, until non-sr units are suppported, strip 'pix' from
+                # unit if it is a per-pixel unit
+                if 'pix' in image_unit.bases:
+                    image_unit = image_unit * u.pix
```

Comment:
should we just leave this out for now if we know we'll immediately remove it?  If so, this whole diff for this method just becomes "flux" -> "sb"

Reply:
if i remove this, it will break for cubes loaded in flux units until i add the logic for converting between square pixels and angles. if thats ok with you i can remove it

Reply:
Either way is fine with me, as long as it gets simplified back to the one liner in the end instead of the double casting to a unit